src/rgw/rgw_op.cc

```diff
@@ -308,7 +308,8 @@ static int read_policy(RGWRados *store,
     obj.init_ns(bucket, oid, mp_ns);
     obj.set_in_extra_data(true);
   } else {
-    obj = rgw_obj(bucket, object);
+    obj = rgw_obj(bucket, object.name);
+    obj.set_instance(s->object.instance);
```

Comment:
@oritwas while this might fix the issue, I'm having trouble understanding what's going on and why it should or shouldn't work. Sounds like there's a deeper problem that we should be fixing.